src/xunit.performance.execution/AllocatedBytesForCurrentThread.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Reflection;
+
+namespace Microsoft.Xunit.Performance.Execution
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    internal static class AllocatedBytesForCurrentThread
+    {
+        /// <summary>
+        /// Loads the GetAllocatedBytesForCurrentThread method.
+        /// </summary>
+        /// <remarks>
+        /// Calling `gcGetAllocMethod.Invoke(null, null);` will cause an object
+        /// to be returned (it allocates a minimum object), so calling the
+        /// method back to back and computing the difference between them will
+        /// not be equals to zero (it will differ by 24 bytes).
+        /// </remarks>
+        static AllocatedBytesForCurrentThread()
+        {
+            var typeInfo = typeof(GC).GetTypeInfo();
+            var method = typeInfo.GetMethod("GetAllocatedBytesForCurrentThread",
+                            BindingFlags.Public | BindingFlags.Static)
+                      ?? typeInfo.GetMethod("GetAllocatedBytesForCurrentThread",
+                            BindingFlags.NonPublic | BindingFlags.Static);
+
+            // TODO: Add meaningful error message.
+
+            _GetAllocatedBytesForCurrentThread = method == null ?
+                (Func<long>)(() => 0xBAAAAAAD) : () => (long)method.Invoke(null, null);
+
+            var allocatedBytesBefore = LastAllocatedBytes;
+            var allocatedBytesAfter = LastAllocatedBytes;
+            s_minAllocatedBytes = allocatedBytesAfter - allocatedBytesBefore;
+        }
+
+        public static long LastAllocatedBytes => _GetAllocatedBytesForCurrentThread.Invoke();
+
+        public static long GetTotalAllocatedBytes(long allocatedBytesBefore, long allocatedBytesAfter)
+        {
+            return allocatedBytesAfter - allocatedBytesBefore; // FIXME: (allocatedBytesAfter - allocatedBytesBefore - s_minAllocatedBytes) sometimes returns a negative value!
```

Comment:
I think we need to investigate and fix this issue before merging.  Otherwise, we are going to have incorrect values being returned.  Per our offline conversation, I expect a trace with benchmark iteration events and all allocations should help us get to the bottom of this.